doc/tutorials/chapter_7/00_sequential_logic.md

```diff
@@ -0,0 +1,291 @@
+
+Then, its time to declare the logic of the module. We want to start with creating a `Sequential()` block, that takes in a clock `clk`, a List of `Conditionals`, and a name for the Sequential (optional).
+
+```dart
+Sequential(clk, []);
+```
+
+For our conditionals, we want to wrap `If.block` that contains a List of `ElseIf` in the `Conditionals`. Note that the `ElseIf` here also mean `Iff` and `Else` that are implemented in ROHD framework. Don't confuse with dart `if` and `else`.
+
+```dart
+If.block();
+```
+
+To build our shift register, we want to say something like:
+
+1. IF reset signal, then data will be set to 0
+2. ELSE swizzle the sin with existing value
+3. SET output port to the data
+
+Our `ShiftRegister` module will be look like this instead.
+
+```dart
+class ShiftRegister extends Module {
+  ShiftRegister(Logic clk, Logic reset, Logic sin,
+      {super.name = 'shift_register'}) {
+    clk = addInput('clk', clk);
+    reset = addInput(reset.name, reset);
+    sin = addInput(sin.name, sin, width: sin.width);
+
+    // output width: Let say, we want 8 bit register
+    const regWidth = 8;
+    final sout = addOutput('sout', width: regWidth);
+
+    // Local signal
+    final data = Logic(name: 'data', width: regWidth); // 0000
+
+    Sequential(clk, [
+      If.block([
+        Iff(reset, [data < 0]),
+        Else([
+          data < [data.slice(regWidth - 2, 0), sin].swizzle() 
+        ])
+      ]),
+    ]);
+
+    sout <= data;
+  }
+}
+```
+
+## ROHD Simulator
+
+Now, it's time to dive in unit test and Simulation module in ROHD. ROHD simulator is a static class accessible as `Simulator` which implements a simple event-based simulator. All `Logic`s in Dart have `glitch` events which propagate values to connected `Logic`s downstream. In this way, ROHD propagates values across the entire graph representation of the hardware (without any `Simulator` involvement required).
+
+The simulator has a concept of (unitless) time, and arbitrary Dart function can be registered to occur ar arbitrary times in the simulator. Asking the `Simulator` to run causes it to iterate through all deposit signals on `logic`s, it propagates values across the hardware.
+
+- To register a function at an arbitrary timestamp, use `Simulator.registerAction`
+
+```dart
+final clk = SimpleClockGenerator(10).clk;
+...
+Simulator.registerAction(25, () => reset.put(0)); // put reset to 0 at time 25
+```
+
+- To set a maximum simulation time, use `Simulator.setMaxSimTime`
+
+```dart
+reset.inject
+...
+Simulator.setMaxSimTime(100); // set maximum time to 100
+```
+
+- To immediately end the simulation at the end of the current timestamp, use `Simulator.endSimulation`
+
+```dart
+Simulator.registerAction(75, Simulator.endSimulation); 
+```
+
+- To run just the next timestamp, use `Simulator.tick`
```

Comment:
Let's dissuade users from calling `Simulator.tick` directly since it can be a little confusing.  There can be multiple `tick`s in one timestamp.  Just don't mention `Simulator.tick`.